src/factors/PriorPose2.jl

```diff
@@ -21,14 +21,30 @@ end
 PriorPose2(x::T) where {T <: IncrementalInference.SamplableBelief} = PriorPose2{T}(x)
 
 function getSample(cfo::CalcFactor{<:PriorPose2}, N::Int=1)
-  return (rand(cfo.factor.Z,N), )
+  Z = cfo.factor.Z
+  M = getManifold(Pose2)
+  p = getPointIdentity(Pose2)
+  
+  Xc = [rand(Z) for _ in 1:N]
+  
+  # X = get_vector.(Ref(M), Ref(p), Xc, Ref(DefaultOrthogonalBasis()))
+  X = hat.(Ref(M), Ref(p), Xc)
+  points = exp.(Ref(M), Ref(p), X)
+
+  return (points, )
```

Comment:
This is fine for SE(2), but as for getSample on Pose2Pose2, we should synchronize on how we want to do it in general.

Reply:
For the transition each factor must have it's own sampler written.  Dont think we need to try make them all use a few functions.  We can just duplicate for now.  

I was not sure how to do BearingRange yet.  I was thinking `ProductManifold(Cirlcle(), Euclidean(1))`.

Reply:
Yes, my current thinking is as a product manifold. But I'm undecided between Circle() and SO(2), perhaps SO(2) because it's implemented as a group.